### DISCUSSION TITLE
Server-side data fetching in Next.js 14 (app router) while Mantine (AppShell) turns everything into client components?

### DISCUSSION THREAD

**Discussion by jnachtigall** [mantinedev/mantine#7087] [Q&A] [answered]
I am using the next-app-template from https://mantine.dev/guides/next/ (Next.js 14 with app router). My `layout.tsx` uses [AppShell](https://mantine.dev/core/app-shell/) which makes [it and all subseqent components a client component](https://mantine.dev/guides/next/#server-components).  (_"Mantine components cannot be used as server components. It means that components will render both on the server and client."_)

My app requires to make a remote backend API call to fetch some json data before it can load its UI. Now it would be useful to do the fetch once at startup as data fetch on the server (and cache there) and not on every browser (client-side fetch). However, as I see it I cannot do server-side data fetching at all using Next.js because Mantine and/or AppShell is used in my layout/page components making every component that comes thereafter to a client component.

Is there a way to server-side fetch data using Next.js server when using Mantine? If so how. 

FWIW, my `app/layout.tsx` looks like the following. How could I server-side fetch in this `layout.tsx` or even better in its sibling `app/page.tsx` or any other component at app startup (on the server)? 

<details>

<summary>app/layout.tsx</summary>

```tsx
'use client';

import '@mantine/core/styles.css';

import { AppShell, ColorSchemeScript, Container, MantineProvider, ScrollArea } from '@mantine/core';
import Footer from '@/components/organisms/Footer/Footer';
import Header from '@/components/organisms/Header/Header';
import { theme } from '../theme';

type Props = {
  children: any;
};

export default function RootLayout({ children }: Readonly<Props>) {
  const banderoleHeight = 48;
  const headerContentHeight = 140;
  const headerHeight = banderoleHeight + headerContentHeight;
  const footerHeight = 200;

  const bodyProps = {
    bg: 'mh2-gray-5',
  };
  return (
    <html lang="de">
      <head>
        <ColorSchemeScript />
        <link rel="shortcut icon" href="/favicon.svg" />
        <meta
          name="viewport"
          content="minimum-scale=1, initial-scale=1, width=device-width, user-scalable=no"
        />
        {/* Name and description have to be placed here because of the 'use client'-directive.
            Using the metadata-export leads to a malfunctioning appshell!*/}
        <meta name="description" content="My description" />
        <title>My Title</title>
      </head>
      <body>
        <MantineProvider theme={theme}>
          <AppShell
            header={{ height: headerHeight }}
            footer={{ height: footerHeight }}
            m="150px 0 0 0"
          >
            <AppShell.Header>
              <Header banderoleHeight={banderoleHeight} headerContentHeight={headerContentHeight} />
            </AppShell.Header>
            <AppShell.Main component={ScrollArea}>
              <Container fluid {...bodyProps}>
                {children}
              </Container>
            </AppShell.Main>
            <AppShell.Footer>
              <Footer height={footerHeight} />
            </AppShell.Footer>
          </AppShell>
        </MantineProvider>
      </body>
    </html>
  );
}

```

</details>

**Comment by rtivital** [comment, marked as answer]
> it and all subseqent components a client component

It is a wrong statement. To learn more about client and server components and how to organize them, see https://www.reddit.com/r/nextjs/comments/1b0k4vz/does_the_child_component_of_a_client_component/

**Reply by jnachtigall** [reply]
Thanks for clarifying and pointing this out. This really helped to find a solution which is: I now have common server-side page.tsx (`export default async function Page() { ...}` which does the `await fetch`. This is with no import from mantine and no `use client`. This then passes the fetched data on as a prop to a "WrappedPage.tsx" component which uses Mantine is a "use client".

**Reply by gentunian** [reply]
@jnachtigall can you elaborate or confirm if I understood this correctly?

```typescript
// some/page.tsx
export default async function Page() {
   const data = await fetch(...)
   
   return (
      <MantineWrapperPage data={data}/>
   )
}

// some/mantine-wrapper-page.tsx
"use client"

// import forms/stuff client side
export default function MantineWrapperPage({data}) {
  // ...
}
```
